### PR TITLE
Letterboxd support

### LINKED ISSUE #30: Feature request: letterboxd integration
Hi,

First and foremost, thanks for making this addon!
I was looking for an easy way to add Radarr integration directly to IMDB, and this fits the bill very nicely. 😄 
(I was originally looking at [web-to-plex](https://github.com/SpaceK33z/web-to-plex) which seems more fully featured, but it looks to be slightly abandoned)

**Feature request**
Support for letterboxd.com. Example link to a film: https://letterboxd.com/film/abelard-die-entmannung/

I started working on a PR for this now, will update here in the coming days

### PR DESCRIPTION
**Pull request by aommm**
Add integration towards Letterboxd, fixes https://github.com/trossr32/sonarr-radarr-lidarr-autosearch-browser-extension/issues/30.

image

image

Tested on MacOS 10.14, Firefox 85, Chrome 87.

**Bonus**
I also added bash scripts for packaging the extension on MacOS/Linux.
- I can confirm that `create_firefox_package.sh` works.
- I'm not 100% sure whether `create_chromium_package.sh` works. I just mimicked the Powershell script which creates a zip file. Not sure what the purpose of a ZIP file is, maybe it's needed when you publish the extension to the store?

Happy to receive any feedback you might have!

### CODE REVIEW

**Review by trossr32 — CHANGES_REQUESTED** [review]
Hi,

Thanks for doing this work, it's really appreciated!

I've tested the code and there's a change I'd like to make. When testing against https://letterboxd.com/film/wonder-woman-1984/ Radarr can't find a moview using the search string 'wonder woman 1984'. It does however work for the terms 'ww84' or 'wonder woman 1984 (2020)'. The easiest way to fix this is to the use the following config for letterboxd in the content_script.js file, which takes the search term from the <meta property="og:title" content="Wonder Woman 1984 (2020)"> tag in the markup's head tag:

```
{
    id: 'letterboxd',
    defaultSite: 'radarr',
    search: {
        containerSelector: 'meta[property="og:title"]',
        selectorType: 'content',
        modifier: null
    },
    match: {
        term: 'letterboxd.com/film/'
    },
    icon: {
        containerSelector: '#featured-film-header > h1',
        locator: 'prepend',
        imgStyles: 'width: 25px; margin: 8px 10px 0 0; float: left;'
    }
}
```

This also removes the need to regex replace newlines in the search term, although I'm happy for that to be left in as it certainly doesn't hurt to have it there.

Additionally, the following switch:

```
switch (integration.search.selectorType) {
    case 'href':
        searchTerm = $(container).attr('href');
        break;

    default: // text
        searchTerm = $(container).text();
        break;
}
```

needs to be replaced with:

```
switch (integration.search.selectorType) {
    case 'href':
        searchTerm = $(container).attr('href');
        break;

    case 'content':
        searchTerm = $(container).attr('content');
        break;

    default: // text
        searchTerm = $(container).text();
        break;
}
```

I'm happy to make the change if you grant me access to your forked repo (https://docs.github.com/en/free-pro-team@latest/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork), or you can do it and update the pull request.

Again, really appreciate the effort you've put in 👍

File: ChromiumExtension/content/js/content_script.js [hunk 1]
```diff
@@ -213,6 +213,29 @@ var settingsPort = chrome.runtime.connect({ name: 'settings' }),
                 locator: 'prepend',
                 imgStyles: 'width: 26px; margin: 0 5px -4px 0;'
             }
+        },
```
Comment (trossr32) [inline ChromiumExtension/content/js/content_script.js:216]:
```
{
    id: 'letterboxd',
    defaultSite: 'radarr',
    search: {
        containerSelector: 'meta[property="og:title"]',
        selectorType: 'content',
        modifier: null
    },
    match: {
        term: 'letterboxd.com/film/'
    },
    icon: {
        containerSelector: '#featured-film-header > h1',
        locator: 'prepend',
        imgStyles: 'width: 25px; margin: 8px 10px 0 0; float: left;'
    }
}

File: FirefoxAddOn/content/js/content_script.js [hunk 2]
```diff
@@ -213,6 +213,29 @@ var settingsPort = browser.runtime.connect({ name: 'settings' }),
                 locator: 'prepend',
                 imgStyles: 'width: 26px; margin: 0 5px -4px 0;'
             }
+        },
```
Comment (trossr32) [inline FirefoxAddOn/content/js/content_script.js:216]:
```
{
    id: 'letterboxd',
    defaultSite: 'radarr',
    search: {
        containerSelector: 'meta[property="og:title"]',
        selectorType: 'content',
        modifier: null
    },
    match: {
        term: 'letterboxd.com/film/'
    },
    icon: {
        containerSelector: '#featured-film-header > h1',
        locator: 'prepend',
        imgStyles: 'width: 25px; margin: 8px 10px 0 0; float: left;'
    }
}

**Comment by aommm** [comment]
Thanks for the feedback, I have made the necessary updates!

Just want to point out that my original solution did work for e.g. [Wonder Woman 1984](https://letterboxd.com/film/wonder-woman-1984/); it would have extracted "Wonder Woman 1984 2020".

But I like your proposed solution more, it is definitely less brittle. (I didn't notice those meta tags myself 🤦)

I left the "strip spaces" code in because it might still be useful down the road.

So this is ready for review again.

**Review by trossr32 — APPROVED** [review]
Excellent, thank you! Approved.

**Pull request merged**